### PR TITLE
Make camera controls script play nice with XR

### PR DESCRIPTION
**Pull request by willeastcott**
* Disable script update when XR session is active
* Restore camera pos/rot when exiting XR

I confirm I have read the [contributing guidelines](https://github.com/playcanvas/engine/blob/master/.github/CONTRIBUTING.md) and signed the [Contributor License Agreement](https://docs.google.com/a/playcanvas.com/forms/d/1Ih69zQfJG-QDLIEpHr6CsaAs6fPORNOVnMv5nuo0cjk/viewform).

### CODE REVIEW

**Review by slimbuck — APPROVED** [review]

**Review by mvaligursky — APPROVED** [review]

**Comment by mvaligursky** [comment]
@willeastcott - good to merge or other plans with this?

**Comment by willeastcott** [comment]
> @willeastcott - good to merge or other plans with this?

Needs testing - I'll do that today.

File: scripts/esm/camera-controls.mjs [hunk 1]
```diff
@@ -315,6 +315,23 @@ class CameraControls extends Script {
...
+        const position = new Vec3();
+        const rotation = new Quat();
+        this.app.xr.on('start', () => {
+            // Store the position and rotation of the camera
+            position.copy(this.entity.getPosition());
+            rotation.copy(this.entity.getRotation());
+
+            // Set the camera's position to the ground
+            const pos = this.entity.getPosition();
+            this.entity.setPosition(pos.x, 0, pos.z);
```
Comment (Maksims) [inline scripts/esm/camera-controls.mjs:328]: XR will override this position on every update, as now there is no parent node, offsetting the camera from origin is only done by moving in XR. So this line won't do anything.
For offsetting position in XR, a parent node is required.

Comment (willeastcott) [reply scripts/esm/camera-controls.mjs]: OK, I have removed that for now. Everything else OK?

Comment (Maksims) [reply scripts/esm/camera-controls.mjs]: This also implies that initial setup of a scene, should have camera at the origin, and content somewhere in front of it, to make sure when entering in XR we are not inside of a content.

Comment (willeastcott) [reply scripts/esm/camera-controls.mjs]: How should we handle that then? I don't want to reintroduce the magical hidden parent node in the `camera-controls.mjs` script again...

Comment (Maksims) [reply scripts/esm/camera-controls.mjs]: Do a proper parent node for the camera entity. And only use it for XR (not camera orbiting). And reset it when exiting XR.

Comment (willeastcott) [reply scripts/esm/camera-controls.mjs]: So I should create a new script called xr-navigation.mjs I guess...

**Pull request merged**